x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+
+import { POST_ACTIONS_CONNECTOR_EXECUTE } from '../../common/constants';
+import {
+  getLangChainMessages,
+  unsafeGetAssistantMessagesFromRequest,
+} from '../lib/langchain/helpers';
+import { buildResponse } from '../lib/build_response';
+import { buildRouteValidation } from '../schemas/common';
+import {
+  PostActionsConnectorExecuteBody,
+  PostActionsConnectorExecutePathParams,
+} from '../schemas/post_actions_connector_execute';
+import { ElasticAssistantRequestHandlerContext } from '../types';
+import { executeCustomLlmChain } from '../lib/langchain/execute_custom_llm_chain';
+
+export const postActionsConnectorExecuteRoute = (
+  router: IRouter<ElasticAssistantRequestHandlerContext>
+) => {
+  router.post(
+    {
+      path: POST_ACTIONS_CONNECTOR_EXECUTE,
+      validate: {
+        body: buildRouteValidation(PostActionsConnectorExecuteBody),
+        params: buildRouteValidation(PostActionsConnectorExecutePathParams),
+      },
+    },
```

Comment:
We'll want to include `tags` in the future so we can limit via RBAC within SecuritySolution. Also, I think we need `access: 'internal',` for serverless. Also, I believe  we'll need an `addVersion` block ala:

```  
.addVersion(
  {
    version: '1',
    validate: {},
   },
```

Reply:
> We'll want to include `tags` in the future so we can limit via RBAC within SecuritySolution. Also, I think we need `access: 'internal',` for serverless. Also, I believe we'll need an `addVersion` block ala:
> 
> ```
> .addVersion(
>   {
>     version: '1',
>     validate: {},
>    },
> ```

I'm wondering if `access: 'internal'` is required for serverless, because the route (in this PR) is registered via `router.post` instead of `router.versioned.post`. Similarly, I'm wondering if `.addVersion` would be applicable to the code in this PR, because it's not using the versioned router.

@YulNaumenko is route registration via `router.post` instead of `router.versioned.post` still valid for serverless, and if so, does this code need to be updated to include `access: 'internal'` in order for it to work with serverless?